File: GameServer/Content/Data/LeagueSandbox-Default/Champions/Global/SummonerFlash.lua
Vector2 = require 'Vector2' -- include 2d vector lib

function onStartCasting()
end

function onFinishCasting()
    local current = Vector2:new(owner.X, owner.Y)
    local to = Vector2:new(spell.X, spell.Y) - current
    local trueCoords
	
    if to:length() > 425 then
        to = to:normalize()
        local range = to * 425
        trueCoords = current:copy() + range
    else
        trueCoords = Vector2:new(spell.X, spell.Y)
    end
	addParticle(owner, "global_ss_flash.troy", owner.X, owner.Y)
    teleportTo(owner, trueCoords.x, trueCoords.y)
	addParticleTarget(owner, "global_ss_flash_02.troy", owner)
end

function applyEffects()
end

function onUpdate(diff)
end

File: Content/LeagueSandbox-Scripts/Champions/Global/SummonerFlash.cs
using System.Numerics;
using GameServerCore.Domain.GameObjects;
using static LeagueSandbox.GameServer.API.ApiFunctionManager;
using LeagueSandbox.GameServer.Scripting.CSharp;
using GameServerCore.Domain.GameObjects.Spell;
using GameServerCore.Domain.GameObjects.Spell.Missile;

namespace Spells
{
    public class SummonerFlash : IGameScript
    {
        public void OnStartCasting(IObjAiBase owner, ISpell spell, IAttackableUnit target)
        {
        }

        public void OnFinishCasting(IObjAiBase owner, ISpell spell, IAttackableUnit target)
        {
            var current = new Vector2(owner.Position.X, owner.Position.Y);
            var spellPos = new Vector2(spell.CastInfo.TargetPosition.X, spell.CastInfo.TargetPosition.Z);
            var to = spellPos - current;
            Vector2 trueCoords;

            if (to.Length() > 425)
            {
                to = Vector2.Normalize(to);
                var range = to * 425;
                trueCoords = current + range;
            }
            else
            {
                trueCoords = spellPos;
            }

            AddParticle(owner, "global_ss_flash.troy", owner.Position);
            TeleportTo(owner, trueCoords.X, trueCoords.Y);
            AddParticleTarget(owner, "global_ss_flash_02.troy", owner);
        }

        public void ApplyEffects(IObjAiBase owner, IAttackableUnit target, ISpell spell, ISpellMissile projectile)
        {
        }

        public void OnUpdate(float diff)
        {
        }

        public void OnActivate(IObjAiBase owner, ISpell spell)
        {
        }

        public void OnDeactivate(IObjAiBase owner, ISpell spell)
        {
        }
    }
}

